### PR TITLE
properly compute if a folder is top level or child extern mounted

### PR DESCRIPTION
**Pull request by mgallien**
asks new permission to server to be able to know if a folder is a top level mounted folder

should allow detecting the top level folders from external storages or group folders

shoudl also make the client reliably detect that it is handling a child folder inside a group folder and be allowed to rename such folders

### CODE REVIEW

File: src/gui/folderstatusmodel.cpp [hunk 1]
```diff
@@ -616,6 +616,7 @@ void FolderStatusModel::fetchMore(const QModelIndex &parent)
     auto props = QList<QByteArray>() << "resourcetype"
                                      << "http://owncloud.org/ns:size"
                                      << "http://owncloud.org/ns:permissions"
+                                     << QByteArrayLiteral("http://nextcloud.org/ns:is-mount-root")
                                      << "http://owncloud.org/ns:fileid";
```
Comment (claucambra) [inline src/gui/folderstatusmodel.cpp:620]: Do we need QByteArrayLiteral here?

File: src/gui/invalidfilenamedialog.cpp [hunk 2]
```diff
@@ -117,7 +117,7 @@ InvalidFilenameDialog::~InvalidFilenameDialog() = default;
 void InvalidFilenameDialog::checkIfAllowedToRename()
 {
     const auto propfindJob = new PropfindJob(_account, QDir::cleanPath(_folder->remotePath() + _originalFileName));
-    propfindJob->setProperties({ "http://owncloud.org/ns:permissions" });
+    propfindJob->setProperties({"http://owncloud.org/ns:permissions", QByteArrayLiteral("http://nextcloud.org/ns:is-mount-root")});
```
Comment (claucambra) [inline src/gui/invalidfilenamedialog.cpp:120]: Do we need QByteArrayLiteral here?

File: src/libsync/caseclashconflictsolver.cpp [hunk 3]
```diff
@@ -220,7 +220,7 @@ void CaseClashConflictSolver::processLeadingOrTrailingSpacesError(const QString
 void CaseClashConflictSolver::checkIfAllowedToRename()
 {
     const auto propfindJob = new PropfindJob(_account, QDir::cleanPath(remoteTargetFilePath()));
-    propfindJob->setProperties({ "http://owncloud.org/ns:permissions" });
+    propfindJob->setProperties({"http://owncloud.org/ns:permissions", QByteArrayLiteral("http://nextcloud.org/ns:is-mount-root")});
```
Comment (claucambra) [inline src/libsync/caseclashconflictsolver.cpp:223]: Do we need QByteArrayLiteral here?

**Review by claucambra — APPROVED** [review]
LGTM, just a pair of small nitpicks

File: src/gui/invalidfilenamedialog.cpp (hunk 2 above)
Comment (mgallien) [reply src/gui/invalidfilenamedialog.cpp:120]: done

File: src/libsync/caseclashconflictsolver.cpp (hunk 3 above)
Comment (mgallien) [reply src/libsync/caseclashconflictsolver.cpp:223]: done

File: src/gui/folderstatusmodel.cpp (hunk 1 above)
Comment (mgallien) [reply src/gui/folderstatusmodel.cpp:620]: done

**Pull request merged**